lib/MetricsProducer.js

```diff
@@ -16,7 +16,7 @@ class MetricsProducer {
         this._kafkaConfig = kafkaConfig;
         this._topic = mConfig.topic;
 
-        this._producer = null;
```

Comment:
Could be cleaner to add an accessor `getProducer()` instead, to keep encapsulation.